src/nvim/edit.c

```diff
@@ -184,6 +184,12 @@ static expand_T compl_xp;
 
 static int compl_opt_refresh_always = FALSE;
 
+static int pum_selected_item = -1;
+static bool pum_request_active = false;
+static int pum_request_wanted_item = -1;
+static bool pum_request_insert = false;
+static bool pum_request_finish = false;
```

Comment:
what's a pum request? would it make sense to name it "pum want" instead (in analogy to "curswant").

It might help navigation/discoverability if the "pum request" states were a flags enum:

```c
enum {
kPumReqActive = 0x001,
kPumReqInsert = 0x002,
kPumReqDone = 0x004,
} PumState
```

Reply:
I guess `want` works. It is replacement for local variables for a function that cannot be written without either restructuring the code, or perhaps refactor the compl data structures to something that can be easily manipulated without code duplication. Why erase the `bool` variables specifically?

Reply:
> Why erase the bool variables specifically?

It's easier (for a human) to "track" one variable than many. It adds some implicit documentation, that all of these states are related.

Reply:
But `wanted_item` is precisely as related, it just happens to be of a different type. One could use a singleton struct `struct { ... } pum_want;`  as a grouping I guess, and avoid unnecessary bit-twiddling.